File: Python_TEA_model/Subsurface TEA Model/ExplorationCost.py
class ExplorationCost:
    """Calculate exploration costs for subsurface project development."""

    def __init__(
        self,
        site_type,
        no_of_exploration_wells,
        acreage,
        ppi_multiplier,
    ):
        self.site_type = str(site_type)
        self.no_of_exploration_wells = int(no_of_exploration_wells)
        self.wellfield_acreage = float(acreage)
        self.ppi_multiplier = float(ppi_multiplier)

    def exploration_cost_calc(self) -> float:
        exploration_permitting_cost = 25_000
        lease_unit_cost = 38
        lease_cost = lease_unit_cost * self.wellfield_acreage

        exp_drilling_permitting = 300_000
        exp_drilling_unit_cost = 150_000
        exp_drilling_cost = exp_drilling_unit_cost * self.no_of_exploration_wells

        if self.site_type == "Greenfield":
            exp_activities_lump_sum = 900_000
        elif self.site_type == "Brownfield":
            exp_activities_lump_sum = 600_000
        else:
            raise ValueError(f"Unsupported site type: {self.site_type}")

        return (
            self.ppi_multiplier * exp_activities_lump_sum
            + exp_drilling_cost
            + exp_drilling_permitting
            + lease_cost
            + exploration_permitting_cost
        )

File: MATLAB_models/Subsurface/ExplorationCost.m
classdef ExplorationCost
    %   ExplorationCost calculates the total cost for pre-drilling permitting,
    %   exploration drilling, non-drilling exploration activities, and acreage
    %   leasing
    %   

    properties
        SiteType
        NoOfExplorationWells
        WellfieldAcreage
        PPIMultiplier
        
    end

    methods
        function Exploration = ExplorationCost(SiteType,NoOfExpWells,Acreage,PPI)
            % Exploration creates instances for the class ExplorationCost
            % using the listed properties
            %   Detailed explanation goes here
            Exploration.SiteType = SiteType; % "Greenfield" or "Brownfield"
            Exploration.NoOfExplorationWells = NoOfExpWells;
            Exploration.WellfieldAcreage = Acreage; % in acres
            Exploration.PPIMultiplier = PPI; % Assumptions are in 2015 dollars (O&G services)
        end

        function TotalExplorationCost = ExplorationCostCalc(Exploration)
            %METHOD1 Summary of this method goes here
            %   Detailed explanation goes here
            % Pre-drilling cost
            ExplorationPermittingCost = 25000; % From discussions with PRM. GETEM uses $70,497

            % Lease cost
            LeaseUnitCost = 38; % $ per acre. Average of the $25-$50 range
            Acreage = Exploration.WellfieldAcreage;
            LeaseCost = LeaseUnitCost * Acreage;

            % Exploration Drilling Permitting
            ExpDrillingPermitting = 300000; % GETEM rounded up

            % Exploration drilling, if needed. From PRM discussions
            ExpDrillingUnitCost = 150000; % per well. Wells are not full-sized wells.
            NumberofExpWells = Exploration.NoOfExplorationWells;
            ExpDrillingCost = ExpDrillingUnitCost * NumberofExpWells;

            %Exploration activities cost from GETEM
            if Exploration.SiteType == "Greenfield"
                ExpActivtiesLumpSum = 900000;
            elseif Exploration.SiteType == "Brownfield"
                ExpActivtiesLumpSum = 600000;
            end
            TotalExplorationCost = (Exploration.PPIMultiplier * ExpActivtiesLumpSum) +...
                ExpDrillingCost + ExpDrillingPermitting + LeaseCost + ExplorationPermittingCost;
        end
    end
end